x-pack/plugins/security_solution/server/lib/detection_engine/rule_schema/model/rule_schemas.ts

```diff
@@ -125,7 +125,7 @@ export const BaseRuleParams = z.object({
   from: RuleIntervalFrom,
   ruleId: RuleSignatureId,
   investigationFields: InvestigationFieldsCombined.optional(),
-  immutable: IsRuleImmutable,
+  immutable: IsRuleImmutable.optional(),
```

Comment:
@jpdjere @banderror this change is causing [the alerting schema checks](https://buildkite.com/elastic/kibana-pull-request/builds/226685#0191344a-e500-4f89-9727-291808d4ae26) to fail, because making this param optional would in fact be a breaking change. 

This change was referenced explicitly [in the RFC](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/docs/rfcs/detection_response/prebuilt_rules_customization.md#internal-rule-schema), but I'm going to revert it for now since:

1. It would be a breaking change to remove this field, and we've already discussed deprecating it
2. All rules currently have this field, and
3. We can easily calculate it for new rules moving forward.

Reply:
https://github.com/elastic/kibana/pull/190198/commits/b5e36aae5dab8345022c70578c465707f4a77d97

Reply:
@rylnd Yes, unfortunately the RFC is outdated in that regard, because at the time of writing we had one idea about how to perform migration-on-write, but eventually it was adjusted because we found some issues with aggregation queries which don't allow us to make the `immutable` field optional in the saved object.

At the rule saved object level, the `immutable` param should continue to be required, and we should always write a correct value to it.